into/backends/h5py.py

```diff
@@ -31,16 +33,8 @@ def discover_h5py_dataset(d):
     return dshape(s.replace('object', 'string'))
 
 
-def varlen_dtype(dt):
-    """ Inject variable length string element for 'O' """
-    if "'O'" not in str(dt):
-        return dt
-    varlen = h5py.special_dtype(vlen=unicode)
-    return np.dtype(eval(str(dt).replace("'O'", 'varlen')))
-
-
```

Comment:
Is this behavior no longer necessary?  It may be that this code is still useful but wasn't appropriately tested.